src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  health = 100;
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore
+      && prey.hidden === false
+      && Animal.alive.includes(prey)
+    ) {
+      prey.health -= 50;
+
+      if (prey.health <= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(prey), 1);
```

Comment:
Will be better to use filter method
Here you used 2 array iterations, with filter it'll be only 1 iteration